revisions.js

```diff
@@ -105,25 +144,89 @@ const classMates = [
 //   return `${member.firstName} ${member.lastName} is ${member.age} years old`;
 // });
 
-// console.log(studentStrings);
+// const arr = classMates.map(student => `${student.firstName} ${student.lastName} is ${student.age} years old.`)
 
-//// To get a single value out of an arry we can use a forEach...
-// let sum = 0; // Accumulator
+// console.log(arr);
 
-// classMates.forEach(student => {
-//   sum += student.age; // sum = sum + student.age
-// });
+// console.log(classMates.reduce((acc, student) => {
+//   return  acc + student.age;
+// }, 0));
 
 // console.log(sum); // -> 367
 
-// ...But Array.prototype.reduce is much better for it
+// const sum = classMates.reduce((total, sum) => total + sum.age, 0);
+// console.log(sum)
+
 
 // Exercise: Use reduce to do the exact same thing that we did above.
 
-const ageSum = classMates.reduce((sum, student) => {
-  return sum + student.age;
-}, 0);
+// const ageSum = classMates.reduce((sum, student) => {
+//   return sum + student.age;
+// }, 0);
+
+// console.log(ageSum); // -> 367
+
+
+// Exercise: Reduce the classMates array to a boolean that indicates whether there is an object with the firstName Mojgan.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+// const includeStudent = classMates.reduce((acc, student)=> {
+//   if(student.firstName === 'Mojgan') acc = true;
+//   return acc;
+// }, false);
+// console.log(includeStudent);
+
+// const includeStudent = classMates.reduce((hasMojgan, student)=> {
+//   return hasMojgan || student.firstName === 'Mojgan';
+// }, false);
+// console.log(includeStudent);
+
+// false || false is false
+// true || false is true; if there is 1 true is not going to check the next
+
+// const includeStudent = classMates.some(v => v.firstName === 'Mojgan')
+// console.log(includeStudent);
+
+// const includeStudent = classMates.filter(v => v.firstName === 'Mojgan')
+// console.log(includeStudent);
```

Comment:
Perfect solution! What if I wanted to check that all the students are over 18? :thinking: Also, I would try to use more descriptive variables, at least at the beginning, it helps you collaborate with other people on your code.